src/data_processing/data_processing_and_training_demo.ipynb

```diff
@@ -0,0 +1,1559 @@
+     },
+     "execution_count": 132,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "print(ppg_train_timeseries_scaled.shape)\n",
+    "ppg_train_timeseries_scaled"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-07-07T21:04:47.426020200Z",
+     "start_time": "2023-07-07T21:04:47.399239400Z"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "see how the second row is just moving first row to the left by 1"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "outputs": [],
+   "source": [
+    "#put ppg and ecg together to form the encoder input with shape (201554,32,2)\n",
+    "encoder_input = np.stack((ppg_train_timeseries_scaled, ecg_train_timeseries_scaled), axis=2)\n",
+    "\n",
+    "#stack the abp as well to form the decoder output with shape (201554,32,1)\n",
+    "decoder_output = np.expand_dims(abp_train_timeseries_scaled, axis=2)\n"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-07-07T14:59:07.111019400Z",
+     "start_time": "2023-07-07T14:59:07.062975700Z"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "outputs": [
+    {
+     "data": {
+      "text/plain": "(201554, 32, 1)"
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "decoder_output.shape"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-07-07T14:59:10.156940300Z",
+     "start_time": "2023-07-07T14:59:10.141150500Z"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "outputs": [],
+   "source": [
+    "#get a dataloader with batch size of 128\n",
+    "from torch.utils.data import TensorDataset, DataLoader\n",
+    "train_data = TensorDataset(torch.from_numpy(encoder_input).float(), torch.from_numpy(decoder_output).float())\n",
+    "train_loader = DataLoader(train_data, batch_size=128, shuffle=True)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "ExecuteTime": {
+     "end_time": "2023-07-07T14:59:17.693628100Z",
+     "start_time": "2023-07-07T14:59:17.649911700Z"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "#### The data preprocessing is finished.\n",
+    "---------------------------------------------------------------------------\n",
+    "Here is a training demo to show that the data loader is working."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Used Haroon's LSTM class"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "outputs": [],
+   "source": [
+    "class lstm_model(nn.Module):\n",

```

Comment:
i think because the lstm class already exists in ../lstm/lstm.py, u can just import lstm from that file and use it here without having to rewrite it from scratch